src/vivarium_gates_iv_iron/components/pregnancy.py

```diff
@@ -228,9 +233,11 @@ def _sample_initial_pregnancy_state_change_date(
     def _sample_pregnancy_outcome_and_duration(
         self,
         is_pregnant: pd.Index,
-        gestational_ages: pd.Series
+        gestational_ages: pd.Series,
+        init: bool = False
     ):
-        p_outcome = self.outcome_probabilities(is_pregnant)
+        # prevent cyclic dependency
+        p_outcome = self.outcome_probabilities.source(is_pregnant) if init else self.outcome_probabilities(is_pregnant)
```

Comment:
We need to be able to call the actual pipeline rather than the source, because otherwise we won't get the impact of hemoglobin on the pregnancy outcome.

Reply:
This is the entire problem of the cyclic dependency. This seems like a necessary tradeoff -- force initial ordering for the very first use then the pipeline with full effects later.